osu.Game/Scoring/ScorePerformanceCache.cs

```diff
@@ -8,14 +8,15 @@
 using osu.Framework.Allocation;
 using osu.Game.Beatmaps;
 using osu.Game.Database;
+using osu.Game.Rulesets.Difficulty;
 
 namespace osu.Game.Scoring
 {
     /// <summary>
     /// A component which performs and acts as a central cache for performance calculations of locally databased scores.
     /// Currently not persisted between game sessions.
     /// </summary>
-    public class ScorePerformanceCache : MemoryCachingComponent<ScorePerformanceCache.PerformanceCacheLookup, double?>
+    public class ScorePerformanceCache : MemoryCachingComponent<ScorePerformanceCache.PerformanceCacheLookup, PerformanceAttributes>
```

Comment:
Not sure about this change. This way every score PP calculation will store a copy of the "perfect play" attribute values. So there'll be N copies of the same data for N user scores, and each copy will be calculated from scratch. Seems kinda wasteful.

Reply:
Would you prefer a separate cache for perfect play performance or computing it on the fly without caching? I don't think a cache is necessary but I could be wrong seeing that `ScorePerformanceCache` is created with `PerformanceStatistic` being its only usage.

Reply:
Caching would be optimal (because you'd only have to cache one per difficulty) but upon checking it seems to be not-horrible performance wise even on marathon maps like https://osu.ppy.sh/beatmapsets/29157#osu/156352 so maybe fine to keep this as is for now.